liblangutil/ParserBase.cpp

```diff
@@ -85,6 +85,45 @@ void ParserBase::expectToken(Token _value, bool _advance)
 		m_scanner->next();
 }
 
+void ParserBase::expectTokenRecoveryDelete(Token _value, bool _advance)
+{
+	Token tok = m_scanner->currentToken();
+	if (tok != _value)
+	{
+		auto tokenName = [this](Token _token)
+		{
+			if (_token == Token::Identifier)
+				return string("identifier");
+			else if (_token == Token::EOS)
+				return string("end of source");
+			else if (TokenTraits::isReservedKeyword(_token))
+				return string("reserved keyword '") + TokenTraits::friendlyName(_token) + "'";
+			else if (TokenTraits::isElementaryTypeName(_token)) //for the sake of accuracy in reporting
+			{
+				ElementaryTypeNameToken elemTypeName = m_scanner->currentElementaryTypeNameToken();
+				return string("'") + elemTypeName.toString() + "'";
+			}
+			else
+				return string("'") + TokenTraits::friendlyName(_token) + "'";
+		};
+
+		Token _token;
+		SourceLocation const& location = SourceLocation{position(), endPosition(), source()};
+		for (_token = m_scanner->currentToken(); _token != _value && _token != Token::EOS; _token = m_scanner->next()) ;
```

Comment:
As of `/CODING_STYLE.md` (point 4), don't put bodies on the same line as the condition.
I would also suggest to write the above a little more explicit:

```cpp
SourceLocation const& location = SourceLocation{position(), endPosition(), source()};
Token token = m_scanner->currentToken(); // no underscore in the beginning (this marks function params only)
while (token != _value && _token != Token::EOS)
    _token = m_scanner->next();
```

Reply:
Change made.